### PR TITLE
feat: Security improvement. Remove uid in Payment Method occ calls urls (GH-2511)

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-2511

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.component.spec.ts [hunk 1]
```diff
@@ -41,9 +41,9 @@ class MockUserService {
   getPaymentMethods(): Observable<PaymentDetails[]> {
     return of([mockPayment]);
   }
-  loadPaymentMethods(_userId: string): void {}
-  deletePaymentMethod(_userId: string, _paymentMethodId: string): void {}
-  setPaymentMethodAsDefault(_userId: string, _paymentMethodId: string): void {}
+  loadPaymentMethods(): void {}
+  deletePaymentMethod(_paymentMethodId: string): void {}
+  setPaymentMethodAsDefault(_paymentMethodId: string): void {}
   get(): Observable<User> {
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.component.spec.ts]: I think we can safely remove this

File: projects/storefrontlib/src/lib/checkout/components/payment-method/payment-method.component.spec.ts [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
   CheckoutStepType,
 } from '../../model/checkout-step.model';
-import { CheckoutConfigService } from '../../checkout-config.service';
-import { defaultStorefrontRoutesConfig } from '../../../ui/pages/default-routing-config';
+import { PaymentMethodComponent } from './payment-method.component';
+
+import createSpy = jasmine.createSpy;
 
 @Component({
   selector: 'cx-icon',
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/lib/checkout/components/payment-method/payment-method.component.spec.ts]: line 59: remove the userid param from the mock function (Can't put review on it so I put it here :P)

**Comment by bgambocjaviniar** [comment]
Note: all of them is working as intended, but I think a follow up ticket is necessary.

Two points:
- When you have multiple payments, and set one payment to default and have already selected another payment during checkout payment details will not show the correct selected payment.
EX: go through the checkout and reach to payment details. Create 2 payments, and have one of them selected. Go to my account and click payment details, but select the other payment as default. Go through checkout and you will see in payment details that the default 'payment' is not selected but the other is still selected.

- When deleting a payment in my account - payment details, there is no global message showing that it was successful removed. In address form, we get that behaviour. 

*Important* Not related to this ticket since this and the previous source code had the same behaviour.

File: projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.component.spec.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-components/myaccount/payment-methods/payment-methods.component.spec.ts]: Fixed.

File: projects/storefrontlib/src/lib/checkout/components/payment-method/payment-method.component.spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/storefrontlib/src/lib/checkout/components/payment-method/payment-method.component.spec.ts]: I agree.  Fixed.

**Comment by plabadie** [comment]
> Note: all of them is working as intended, but I think a follow up ticket is necessary.
> 
> Two points:
> 
> * When you have multiple payments, and set one payment to default and have already selected another payment during checkout payment details will not show the correct selected payment.
>   EX: go through the checkout and reach to payment details. Create 2 payments, and have one of them selected. Go to my account and click payment details, but select the other payment as default. Go through checkout and you will see in payment details that the default 'payment' is not selected but the other is still selected.
> * When deleting a payment in my account - payment details, there is no global message showing that it was successful removed. In address form, we get that behaviour.
> 
> _Important_ Not related to this ticket since this and the previous source code had the same behaviour.


For the first point, I'm not sure what the expected result should be, but maybe it has the expected behaviour already.  If a payment method is associated to the cart, I would not necessarily expect it to change to the default payment method if the default changes after a payment method is selected for the cart.  In your use case, I don't think the cart information is reset because the user left checkout process to se a fault payment method.  You basically restart checkout with information already selected from the first attempt.

Second point, good observation.  It needs improvement.

Since these are not regression introduced by this PR, please create follow-up tickets for these 2 concerns so we can get to the bottom of it.

**Pull request merged**